jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -161,4 +189,200 @@ public String asString()
+            throw new IllegalArgumentException("Bad cookie name");
+
+        // Format value and params
+        StringBuilder buf = new StringBuilder();
+
+        // Name is checked for legality by servlet spec, but can also be passed directly so check again for quoting
+        boolean quote_name=isQuoteNeededForCookie(_name);
+        quoteOnlyOrAppend(buf,_name,quote_name);
+
+        buf.append('=');
+
+        // Append the value
+        boolean quote_value=isQuoteNeededForCookie(_value);
+        quoteOnlyOrAppend(buf,_value,quote_value);
+
+        // Look for domain and path fields and check if they need to be quoted
+        boolean has_domain = _domain!=null && _domain.length()>0;
+        boolean quote_domain = has_domain && isQuoteNeededForCookie(_domain);
+        boolean has_path = _path!=null && _path.length()>0;
+        boolean quote_path = has_path && isQuoteNeededForCookie(_path);
+
+        // Upgrade the version if we have a comment or we need to quote value/path/domain or if they were already quoted
+        int version = _version;
+        if (version==0 && ( _comment!=null || quote_name || quote_value || quote_domain || quote_path ||
+            QuotedStringTokenizer.isQuoted(_name) || QuotedStringTokenizer.isQuoted(_value) ||
+            QuotedStringTokenizer.isQuoted(_path) || QuotedStringTokenizer.isQuoted(_domain)))
+            version=1;
+
+        // Append version
+        if (version==1)
+            buf.append (";Version=1");
+        else if (version>1)
+            buf.append (";Version=").append(version);
+
+        // Append path
+        if (has_path)
+        {
+            buf.append(";Path=");
+            quoteOnlyOrAppend(buf,_path,quote_path);
+        }
+
+        // Append domain
+        if (has_domain)
+        {
+            buf.append(";Domain=");
+            quoteOnlyOrAppend(buf,_domain,quote_domain);
+        }
+
+        // Handle max-age and/or expires
+        if (_maxAge >= 0)
+        {
+            // Always use expires
+            // This is required as some browser (M$ this means you!) don't handle max-age even with v1 cookies
+            buf.append(";Expires=");
+            if (_maxAge == 0)
+                buf.append(__01Jan1970_COOKIE);
+            else
+                DateGenerator.formatCookieDate(buf, System.currentTimeMillis() + 1000L * _maxAge);
+
+            // for v1 cookies, also send max-age
+            if (version>=1)
+            {
+                buf.append(";Max-Age=");
+                buf.append(_maxAge);
+            }
+        }
+
+        // add the other fields
+        if (_secure)
+            buf.append(";Secure");
+        if (_httpOnly)
+            buf.append(";HttpOnly");
+        if (_comment != null)
+        {
+            buf.append(";Comment=");
+            quoteOnlyOrAppend(buf,_comment,isQuoteNeededForCookie(_comment));
+        }
+        return buf.toString();
+    }
+
+    public String getRFC6265SetCookie()
+    {
```

Comment:
Is there some more refactoring that could be done between the getRFC2965SetCookie and getRFC6265SetCookie methods? The method bodies look almost (but not quite) identical.

Reply:
mandatory spaces, lack of quoting, and different syntax checks are the biggest differences I can see.

Reply:
I had a look and there is would be too difficult and error prone to merge these.